File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.cpp
class Solution {
public:
    int xorAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
       
        int MOD = 1000000007;
        
        // Process each query
        for (const auto& q : queries) {
            int l = q[0], r = q[1], k = q[2], v = q[3];
            for (int idx = l; idx <= r; idx += k) {
                // 1LL casts nums[idx] to a 64-bit long long to prevent overflow during multiplication
                nums[idx] = (1LL * nums[idx] * v) % MOD;
            }
        }
        
        // Calculate the bitwise XOR of the modified array
        int ans = 0;
        for (int num : nums) {
            ans ^= num;
        }
        
        return ans;
    }
};

File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.py
class Solution(object):
    def xorAfterQueries(self, nums, queries):
        """
        :type nums: List[int]
        :type queries: List[List[int]]
        :rtype: int
        """
        MOD = 10**9 + 7
        
        # Process each query
        for l, r, k, v in queries:
            for idx in range(l, r + 1, k):
                nums[idx] = (nums[idx] * v) % MOD
                
        # Calculate the bitwise XOR of the modified array
        ans = 0
        for num in nums:
            ans ^= num
            
        return ans